notebooks/curation_api/python/src/dataset.py

```diff
@@ -1,18 +1,103 @@
+import json
+
 import boto3
 import logging
 import os
+import re
 import requests
 import threading
 from botocore.credentials import RefreshableCredentials
 from botocore.session import get_session
 from datetime import datetime, timezone
+from typing import Tuple
 
 from src.utils.logger import get_custom_logger
 from src.utils.http import url_builder, get_headers
 
 
 logger = get_custom_logger()
 
+UUID_REGEX = r"[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
+DATASET_ID_REGEX = f"(?P<dataset_uuid>{UUID_REGEX})"
+CURATOR_TAG_PREFIX_REGEX = r"(?P<tag_prefix>.*)"
+EXTENSION_REGEX = r"(?P<extension>h5ad)"
+
+
+def get_identifier_type_and_value(identifier: str) -> Tuple[str, str]:
```

Comment:
What do you think of adding a parameter to both delete and patch dataset asking the user to input whether the current identifier they're providing is a curator_tag or dataset_uuid? Presumably, they know the difference and can save us from doing this work on the backend

Reply:
we can add a quick check that their input matches either of the available options, else raise an exception asking for a proper input

Reply:
>Presumably, they know the difference and can save us from doing this work on the backend

We can _determine_ the difference and can therefore save _them_ from the hassle on the client side 😉 

But seriously, I think it's better this way, just like when they pass us a Collection uuid we don't want to make them also tell us whether it's a private or public Collection. It's unnecessary work for them, and it'll lead to accidental errors when they change an identifier and forget to change the type that they would need to pass in as well.

Reply:
> We can determine the difference and can therefore save them from the hassle on the client side

+1. Curator time is valuable.